### PR TITLE
[REF] check_xml

### PR DESCRIPTION
**Pull request by Xavier-Do**
- additional validation in check_xml for views + corresponding fixes
- performance improvement of read combined

Install of all modules average time goes from  ~8m to ~6m31
(This time improvement takes the fix of two accessibility checks (fa title and stuff) that where removed in a #39494 and replaced in this pr)

### CODE REVIEW

File: addons/account/views/res_config_settings_views.xml [hunk 1]
```diff
@@ -54,7 +54,7 @@
                                     <div class="content-group">
                                         <div class="row mt16">
                                             <label string="Sales Tax" for="sale_tax_id" class="col-lg-3 o_light_label"/>
-                                            <field name="sale_tax_id" domain="[('type_tax_use', 'in', ('sale', 'all')), ('company_id', '=', company_id)]"/>
+                                            <field name="sale_tax_id" domain="[('type_tax_usea', 'in', ('sale', 'all')), ('company_ida', '=', company_id)]"/>
```
Comment (mreficent) [inline addons/account/views/res_config_settings_views.xml]: ??

Comment (Xavier-Do) [reply addons/account/views/res_config_settings_views.xml]: creating failures to test new  checks.

File: addons/crm/views/crm_lead_views.xml [hunk 2]
```diff
@@ -208,7 +208,7 @@
                             </group>
                             <group>
                                 <field name="user_id" domain="[('share', '=', False)]"
-                                    context="{'default_groups_ref': ['base.group_user', 'base.group_partner_manager', 'sales_team.group_sale_salesman_all_leads'], 'team_id': team_id}"/>
```
Comment (tde-banana-odoo) [inline addons/crm/views/crm_lead_views.xml]: Oooh remaining of our famous "create users with some default groups" ... snif. Good ol' openerp.

File: odoo/addons/base/tests/test_views.py [hunk 3]
```diff
@@ -2288,3 +2466,40 @@ def test_render(self):
...
+
+    def test_process_value(self):
+        self.assertEqual(
+            view_validation.process_value_str("context_today().strftime('%Y-%m-%d')"),
+            []
+        )
+        self.assertEqual(
+            view_validation.process_value_str("field or not field2"),
+            ['field', 'field2']
+        )
```
Comment (rco-odoo) [inline odoo/addons/base/tests/test_views.py]: what about multiple occurrences, e.g., `"foo and foo[0] or False"`

File: odoo/addons/base/models/res_users.py [hunk 4]
```diff
@@ -1103,7 +1112,10 @@ def _update_user_groups_view(self):
             new_context = dict(view._context)
             new_context.pop('install_mode_data', None)  # don't set arch_fs for this computed view
             new_context['lang'] = None
-            view.with_context(new_context).write({'arch': xml_content})
+            if xml_content != view.arch: # avoid usefull xml validation if no change
```
Comment (rco-odoo) [inline odoo/addons/base/models/res_users.py]:
```suggestion
            if xml_content != view.arch: # avoid useless xml validation if no change
```

Comment (rco-odoo) [reply odoo/addons/base/models/res_users.py]: and the 'if' may be above the definition of `new_context`

File: odoo/addons/base/models/res_partner.py [hunk 5]
```diff
@@ -61,7 +61,7 @@ def _fields_view_get_address(self, arch):
                 #(e.g fields not present on the model). In that case we just return arch
                 if self._name != 'res.partner':
                     try:
-                        self.env['ir.ui.view'].postprocess_and_fields(self._name, sub_view_node, None)
+                        self.env['ir.ui.view'].postprocess_and_fields(self._name, sub_view_node, None, validate=False)
```
Comment (rco-odoo) [inline odoo/addons/base/models/res_partner.py]: Beware: here the purpose is to validate the view!

Comment (Xavier-Do) [reply odoo/addons/base/models/res_partner.py]: I'm not sure, validate is similar to check_field_names context key that wasn't given here before.

File: addons/website_theme_install/models/website.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# -*- coding: utf-8 -*-
```
Comment (mreficent) [inline addons/website_theme_install/models/website.py]: this module has been merged into website in https://github.com/odoo/odoo/commit/88e910e187346c095a6f0eeb8be647da1e391035

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

File: addons/mrp/views/mrp_workorder_views.xml [hunk 7]
```diff
@@ -155,7 +155,7 @@
                         </group>
                     </group>
                     <h4 attrs="{'invisible': [('raw_workorder_line_ids', '=', [])]}">Components</h4>
-                    <field name="raw_workorder_line_ids" attrs="{'invisible': [('raw_workorder_line_ids', '=', [])], 'readonly': [('state', 'in', 'cancel', 'done')]}">
+                    <field name="raw_workorder_line_ids" attrs="{'invisible': [('raw_workorder_line_ids', '=', [])], 'readonly': [('state', 'in', ('cancel', 'done'))]}">
```
Comment (mreficent) [inline addons/mrp/views/mrp_workorder_views.xml]: will this commit of "various fixes in xml views" be backported (full or partially)? 🤞

File: odoo/addons/base/tests/__init__.py [hunk 8]
```diff
@@ -35,7 +35,6 @@
 from . import test_translate
 #import test_uninstall  # loop
 from . import test_user_has_group
-from . import test_view_validation
```
Comment (mreficent) [inline odoo/addons/base/tests/__init__.py]: what happens with the `/tests/test_view_validation.py` file? why not delete it? 🤔

File: addons/mrp/views/mrp_workorder_views.xml (hunk 7 above)
Comment (rco-odoo) [reply addons/mrp/views/mrp_workorder_views.xml]: @mreficent some of these fixes have or will be backported, but not systematically.

File: odoo/addons/base/tests/__init__.py (hunk 8 above)
Comment (rco-odoo) [reply odoo/addons/base/tests/__init__.py]: @mreficent thanks for pointing that out!

**Comment by rco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**